src/storage/compactor/src/lib.rs

```diff
@@ -16,79 +16,37 @@ mod compactor_observer;
 mod rpc;
 mod server;
 
-use clap::Parser;
-
-use crate::server::compactor_serve;
-
-/// Command-line arguments for compute-node.
-#[derive(Parser, Clone, Debug)]
-pub struct CompactorOpts {
-    // TODO: rename to listen_address and separate out the port.
-    #[clap(long, default_value = "127.0.0.1:6660")]
-    pub host: String,
-
-    // Optional, we will use listen_address if not specified.
-    #[clap(long)]
-    pub client_address: Option<String>,
-
-    // TODO: This is currently unused.
-    #[clap(long)]
-    pub port: Option<u16>,
-
-    #[clap(long, default_value = "")]
-    pub state_store: String,
-
-    #[clap(long, default_value = "127.0.0.1:1260")]
-    pub prometheus_listener_addr: String,
-
-    #[clap(long, default_value = "0")]
-    pub metrics_level: u32,
-
-    #[clap(long, default_value = "http://127.0.0.1:5690")]
-    pub meta_address: String,
-
-    /// It's a hint used by meta node.
-    #[clap(long, default_value = "16")]
-    pub max_concurrent_task_number: u64,
-
-    #[clap(long)]
-    pub compaction_worker_threads_number: Option<usize>,
-
-    /// The path of `risingwave.toml` configuration file.
-    ///
-    /// If empty, default configuration values will be used.
-    ///
-    /// Note that internal system parameters should be defined in the configuration file at
-    /// [`risingwave_common::config`] instead of command line arguments.
-    #[clap(long, default_value = "")]
-    pub config_path: String,
-}
-
 use std::future::Future;
 use std::pin::Pin;
 
-pub fn start(opts: CompactorOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
+use risingwave_common::config::{load_config, CompactorConfig};
+
+use crate::server::compactor_serve;
+
+pub fn start(opts: CompactorConfig) -> Pin<Box<dyn Future<Output = ()> + Send>> {
```

Comment:
When will this `opts` be used? 👀

Reply:
- To get the path of the config file
- To overwrite the config read from the file